lib/SPIRV/LLVMToSPIRVDbgTran.cpp

```diff
@@ -988,8 +988,18 @@ SPIRVEntry *LLVMToSPIRVDbgTran::transDbgFunction(const DISubprogram *Func) {
 
     if (DISubprogram *FuncDecl = Func->getDeclaration())
       Ops.push_back(transDbgEntry(FuncDecl)->getId());
-    else
+    else {
       Ops.push_back(getDebugInfoNoneId());
+      if (BM->getDebugInfoEIS() == SPIRVEIS_NonSemantic_Kernel_DebugInfo_100) {
+        // Translate targetFuncName mostly for Fortran trampoline function if it
+        // is the case
+        StringRef TargetFunc = Func->getTargetFuncName();
+        if (!TargetFunc.empty()) {
+          Ops.resize(MinOperandCount + 2);
+          Ops[TargetFunctionNameIdx] = BM->getString(TargetFunc.str())->getId();
```

Comment:
Why not to use just `push_back` here? Not obvious why the `resize` is needed here

Reply:
`push_back` can be used here indeed. I was considering it and logic to prefer resize was the following:
We need to add `TargetFunctionNameIdx` in `lib/SPIRV/libSPIRV/SPIRV.debug.h` anyway to reuse it during reverse translation. So for the consistency with the code around it make sense to reuse it in forward translation as well. Also `resize` was also used on line 976 as well in similar example.

Reply:
Sorry, i'm not following it fully - seems that resize to `MinOperandCount+1` or `TargetFunctionNameIdx` would be enough.

Also, I found that we didn't define `Ops[DeclarationIdx]` here, but the spec says:

> When Target function present and is not an <id> of [DebugInfoNone](https://github.com/KhronosGroup/SPIRV-Registry/blob/4fab24de6c61ae302fe5c3ba900fcddeae349a4a/nonsemantic/NonSemantic.Kernel.DebugInfo.100.asciidoc#DebugInfoNone) - Declaration must be an <id> of [DebugInfoNone](https://github.com/KhronosGroup/SPIRV-Registry/blob/4fab24de6c61ae302fe5c3ba900fcddeae349a4a/nonsemantic/NonSemantic.Kernel.DebugInfo.100.asciidoc#DebugInfoNone).

Am I right that we should fill this Declaration Id any time we translate Target Id?

Reply:
>> Sorry, i'm not following it fully - seems that resize to MinOperandCount+1 or TargetFunctionNameIdx would be enough.

Nope, because there is actually one `push_back` earlier in the code :) See line 992 `Ops.push_back(getDebugInfoNoneId());` , it address both of the questions.

Reply:
replaced resize with push_back